xarray/core/rolling.py

```diff
@@ -202,27 +228,22 @@ def reduce(self, func, **kwargs):
         reduced : DataArray
             Array with summarized data.
         """
-
-        windows = [window.reduce(func, dim=self.dim, **kwargs)
-                   for _, window in self]
-
-        # Find valid windows based on count
-        if self.dim in self.obj.coords:
-            concat_dim = self.window_labels
-        else:
-            concat_dim = self.dim
-        counts = concat([window.count(dim=self.dim) for _, window in self],
-                        dim=concat_dim)
-        result = concat(windows, dim=concat_dim)
-        # restore dim order
-        result = result.transpose(*self.obj.dims)
-
-        result = result.where(counts >= self._min_periods)
-
-        if self.center:
-            result = self._center_result(result)
-
-        return result
+        fill_value = dtypes.reduceable_fill_value(self.obj.dtype)
```

Comment:
I'm not sure this makes sense in the long term. This feels like a work-around for how we don't support `skipna=True` for object dtypes, which leads to an inconsistency with pandas:
```
>>> import xarray
>>> da = xarray.DataArray([True, False], [('x', [0, 1])]).reindex(x=[0, 1, 2])
>>> da
<xarray.DataArray (x: 3)>
array([True, False, nan], dtype=object)
Coordinates:
  * x        (x) int64 0 1 2

>>> da.to_series()
x
0     True
1    False
2      NaN
dtype: object

>>> da.to_series().sum()
1

>>> da.sum()
<xarray.DataArray ()>
array(nan)

>>> da.sum(skipna=True)
NotImplementedError: skipna=True not yet implemented for sum with dtype object
```

Reply:
I think `np.nansum(da.values)` works for `DataArray` in your example, but why do we decided not to support it for `object` dtype?
https://github.com/pydata/xarray/blob/009291139fde0c859ee565141cdb3b6a3d28cba0/xarray/core/duck_array_ops.py#L188-L192 
Also, do you know how pandas supports this?

Reply:
It looks like the problematic op was `nanmean`, though there might be others, too:
```
>>> np.nanmean(da.values)
AttributeError: 'int' object has no attribute 'dtype'
```

I suppose this is why pandas has its own nanmean implementation:
https://github.com/pandas-dev/pandas/blob/d3f7d2a666aa824e2df98083aa5c1fd9bb63252e/pandas/core/nanops.py#L356

Reply:
Thanks. 
A tentative workaround would be to support `skipna=True` for `nansum` (and `nanmax`, etc..), but not for `nanmean`.
I guess we can import pandas nan-reduce methods in the future.